### PR TITLE
Add package to {cpus,uncores} matching functions

### PR DESCRIPTION
**Pull request by sgruszka**
Allow to specify cpus and uncores based on package (socket) number.

### CODE REVIEW

File: tuned/profiles/functions/function_package2cpus.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+SYSFS_DIR = "/sys/devices/system/cpu/"
+
+class package2cpus(base.Function):
+	"""
+	Provides cpu device list for a package (socket)
+	"""
+
+	def __init__(self):
+		# 1 argument
```
Comment (yarda) [inline tuned/profiles/functions/function_package2cpus.py:17]: Could it make sense to have 1 to infinite arguments? E.g. If I want to get devices for package 1 and 2:
```ini
devices=${f:package2cpus:1:2}
```
If such queries make sense, the constrain could be relaxed and the arguments processed in the loop.

File: tuned/profiles/functions/function_package2uncores.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+SYSFS_DIR = "/sys/devices/system/cpu/intel_uncore_frequency/"
+
+class package2uncores(base.Function):
+	"""
+	Provides uncore device list for a package (socket)
+	"""
+
+	def __init__(self):
+		# 1 argument
```
Comment (yarda) [inline tuned/profiles/functions/function_package2uncores.py:17]: Again, maybe relax maximal number of arguments? If it makes sense.

File: tuned/profiles/functions/function_package2uncores.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+					continue
+			else:
+				# uncore string is in form package_NN_die_MM
+				# TODO make this more reliable?
+				value = uncore[8:10]
+
+			try:
+				package_id = int(value)
+			except ValueError:
+				continue
```
Comment (yarda) [inline tuned/profiles/functions/function_package2uncores.py:63]: At least log.debug or maybe log.info/warn would be useful here, e.g. to detect case if kernel changes the interface.

File: tuned/profiles/functions/function_package2cpus.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+				continue
+
+			value = cmd.read_file(f)
+			if len(value) == 0:
+				continue
+
+			try:
+				package_id = int(value)
+			except ValueError:
+				continue
```
Comment (yarda) [inline tuned/profiles/functions/function_package2cpus.py:50]: I think at least log.debug could be useful here.

**Comment by yarda** [comment]
Is it ready for merge, or still WIP (because it has WIP flag)? From my point of view it's ready. I added few comments that could improve it, but I can see no problem with this PR.

**Comment by sgruszka** [comment]
> Is it ready for merge, or still WIP (because it has WIP flag)? From my point of view it's ready. I added few comments that could improve it, but I can see no problem with this PR.

Addressed the comments and remove WIP flag. Please review.

File: tuned/profiles/functions/function_package2cpus.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+			f = SYSFS_DIR + cpu + "/topology/physical_package_id"
+			if not os.path.exists(f):
+				log.warning("File '%s' does not exist" % f)
+				continue
+
+			try:
+				value = cmd.read_file(f)
+				package_id = int(value)
+			except ValueError:
+				log.warning("Failed to get package id for cpu '%s'" % cpu)
```
Comment (yarda) [inline tuned/profiles/functions/function_package2cpus.py:45]: I think you could also log the content of the `value` which may help with debugging possible problems.

File: tuned/profiles/functions/function_package2uncores.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+				value = cmd.read_file(f)
+			else:
+				# uncore string is in form: package_NN_die_MM, do not expect more than 99 packages
+				value = uncore[8:10]
+
+			try:
+				package_id = int(value)
+			except ValueError:
+				log.warning("Failed to get package id for uncore '%s'" % uncore)
```
Comment (yarda) [inline tuned/profiles/functions/function_package2uncores.py:58]: I think you could also log the content of the `value` which may help with debugging possible problems.

**Review by yarda — COMMENTED** [review]
It's not blocker, but I think it's worth fixing.

**Comment by yarda** [comment]
CI failure seems unrelated to this PR.

**Comment by sgruszka** [comment]
Addressed the comments, please review.

**Review by yarda — APPROVED** [review]
LGTM, thanks.

CI failures are unrelated:
- Centos 7 CI failure is caused by dropped EPEL-7, it's being worked on.
- Fedora rawhide failure is caused by systemd rate limiting, workaround will be enabled in beakerlib/tuned library.

**Pull request merged**